### PR TITLE
[SQSERVICES-1643] Servantify brig account API 3 - `POST /activate`

### PR DESCRIPTION
**Pull request by battermann**
https://wearezeta.atlassian.net/browse/SQSERVICES-1643

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: libs/wire-api/src/Wire/API/User/Activation.hs [hunk 1]
```diff
@@ -80,14 +81,42 @@ instance ToByteString ActivationTarget where
...
+        keyDocs = description ?~ "An opaque key to activate, as it was sent by the API."
+        phoneDocs = description ?~ "A known phone number to activate."
+        emailDocs = description ?~ "A known email address to activate."
+
+    maybeActivationKeyTargetFromTuple :: (Maybe ActivationKey, Maybe Phone, Maybe Email) -> Parser ActivationTarget
+    maybeActivationKeyTargetFromTuple = \case
+      (Just key, _, _) -> pure $ ActivateKey key
+      (_, _, Just email) -> pure $ ActivateEmail email
+      (_, Just phone, _) -> pure $ ActivatePhone phone
+      _ -> fail "key, email or phone must be present"
```
Comment (fisx) [inline libs/wire-api/src/Wire/API/User/Activation.hs:112]:
```suggestion
maybeActivationKeyTargetFromTuple :: (Maybe ActivationKey, Maybe Phone, Maybe Email) -> Parser ActivationTarget
maybeActivationKeyTargetFromTuple = \case
  (Just key, _, _) -> pure $ ActivateKey key
  (_, _, Just email) -> pure $ ActivateEmail email
  (_, Just phone, _) -> pure $ ActivatePhone phone
  _ -> fail "key, email or phone must be present"
```

I like symmetry!  :)

File: libs/wire-api/src/Wire/API/User/Activation.hs [hunk 2]
```diff
@@ -117,54 +146,24 @@ data Activate = Activate
...
-    Doc.optional
-  Doc.property "email" Doc.string' $ do
-    Doc.description "A known email address to activate."
-    Doc.optional
-  Doc.property "phone" Doc.string' $ do
-    Doc.description "A known phone number to activate."
-    Doc.optional
-  Doc.property "code" Doc.string' $
-    Doc.description "The activation code."
-  Doc.property "label" Doc.string' $ do
```
Comment (fisx) [inline libs/wire-api/src/Wire/API/User/Activation.hs:135]: does was a lie before, right?  the json instances didn't know about this field?

File: libs/wire-api/src/Wire/API/User/Activation.hs [hunk 3]
```diff
@@ -80,14 +81,42 @@ instance ToByteString ActivationTarget where
...
+  deriving newtype (ToSchema, ToByteString, FromByteString, ToJSON, FromJSON, Arbitrary)
 
 instance ToParamSchema ActivationKey where
   toParamSchema _ = S.toParamSchema (Proxy @Text)
 
 instance FromHttpApiData ActivationKey where
   parseUrlPiece = fmap ActivationKey . parseUrlPiece
 
+maybeActivationKeyObjectSchema :: Schema.ObjectSchemaP Schema.SwaggerDoc (Maybe ActivationKey, Maybe Phone, Maybe Email) ActivationTarget
+maybeActivationKeyObjectSchema =
```
Comment (fisx) [inline libs/wire-api/src/Wire/API/User/Activation.hs:93]: add a description somewhere in english that the 3 are mutually exclusive?  (i don't think there is a way in swagger2 to express that.)

**Review by fisx — APPROVED** [review]

File: libs/wire-api/src/Wire/API/User/Activation.hs (hunk 2 above)
Comment (battermann) [reply libs/wire-api/src/Wire/API/User/Activation.hs:135]: Yeah, I noticed that, too.

**Pull request merged**